core/src/apps/cardano/auxiliary_data.py

```diff
@@ -0,0 +1,255 @@
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.messages import (
+    CardanoAddressType,
+    CardanoAuxiliaryDataType,
+    CardanoMetadataType,
+)
+
+from apps.common import cbor
+
+from ..common.seed import remove_ed25519_prefix
+from .address import derive_address_bytes, derive_human_readable_address
+from .helpers import INVALID_AUXILIARY_DATA, bech32
+from .helpers.bech32 import HRP_PUBLIC_KEY
+from .helpers.paths import SCHEMA_STAKING_ANY_ACCOUNT
+from .layout import confirm_catalyst_registration, show_auxiliary_data_hash
+
+if False:
+    from typing import Dict, List, Optional, Tuple, Union
+    from trezor import wire
+
+    from trezor.messages.CardanoCatalystRegistrationParametersType import (
+        CardanoCatalystRegistrationParametersType,
+    )
+    from trezor.messages.CardanoTxAuxiliaryDataType import CardanoTxAuxiliaryDataType
+
+    CatalystRegistrationBody = Dict[int, Union[bytes, int]]
+    CatalystRegistrationBodyMap = Dict[int, CatalystRegistrationBody]
+    CatalystRegistrationSignatureMap = Dict[int, bytes]
+    SignedVotingRegistrationMap = Dict[
```

Comment:
`SignedCatalystRegistrationMap` would be more consistent respective to the other names